frontend/src/components/common/header/components/search-bar/search-bar.tsx

```diff
@@ -0,0 +1,56 @@
+import { FC } from 'common/types/types';
+import { Icon } from 'components/common/common';
+import { Input } from 'components/common/input/input';
+import { debounce } from 'helpers/helpers';
+import {
+  useAppDispatch,
+  useAppForm,
+  useEffect,
+  useFormControl,
+} from 'hooks/hooks';
+import { dashboardActions } from 'store/actions';
+
+import { DEFAULT_SEARCH_PAYLOAD, SEARCH_DELAY_MS } from './common/constants';
+import { SearchPayload } from './common/types/types';
+import styles from './styles.module.scss';
+
+const SearchBar: FC = () => {
+  const dispatch = useAppDispatch();
+  const onSearch = (search: string): void => {
+    dispatch(dashboardActions.getCourses({ title: search, categoryKey: '' }));
+  };
+
+  const { control, errors } = useAppForm<SearchPayload>({
+    defaultValues: DEFAULT_SEARCH_PAYLOAD,
+  });
+
+  const { field } = useFormControl({ name: 'search', control: control });
+  const { value } = field;
+
+  const handleSearch = (): void => onSearch(value);
```

Comment:
```suggestion
  const handleSearch = (): void => {
      dispatch(dashboardActions.getCourses({ title: search, categoryKey: '' }));
  }
```
And no need for `onSearch`